compiler/test/fail_compilation/discard_value.d

```diff
@@ -0,0 +1,34 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/discard_value.d(24): Error: the result of the equality expression `3 is 3` is discarded
+fail_compilation/discard_value.d(25): Error: the result of the equality expression `null !is null` is discarded
+fail_compilation/discard_value.d(26): Error: the result of the equality expression `v == 0` is discarded
+fail_compilation/discard_value.d(27): Error: the result of the equality expression `v == 0` is discarded
+fail_compilation/discard_value.d(28): Error: `!__equals("", "")` has no effect
```

Comment:
Ideally, we would not spit internal names on the user.

Reply:
I agree, but this error stems from somewhere else than sideffect.d, so I'd leave that as future enhancement.